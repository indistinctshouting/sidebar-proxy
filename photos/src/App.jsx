import React from 'react';
import $ from 'jquery';
import Photos from './Photos.jsx';
import app from './../public/app.css';

import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowUp } from '@fortawesome/free-solid-svg-icons';
import { faArrowDown } from '@fortawesome/free-solid-svg-icons';
import { faTruck } from '@fortawesome/free-solid-svg-icons';
import { faThLarge } from '@fortawesome/free-solid-svg-icons';
import { faShareSquare } from '@fortawesome/free-solid-svg-icons';
import { faFlag } from '@fortawesome/free-solid-svg-icons';

library.add(faArrowUp);
library.add(faArrowDown);
library.add(faTruck);
library.add(faThLarge);
library.add(faShareSquare);
library.add(faFlag);


class App extends React.Component {
  constructor() {
    super();

    this.state = {
      images: [],
    };

    this.getData = this.getData.bind(this);
    this.nextPhoto = this.nextPhoto.bind(this);
    this.previousPhoto = this.previousPhoto.bind(this);
    this.timedScroll = window.setInterval(this.nextPhoto, 8000);
  }

  componentDidMount() {
    this.getData();
  }

  getData() {
    var context = this;
    var foodPics = [];
    const restId = window.location.pathname.split('/')[1];
    var getRandomInt = function(min, max) {
      return Math.floor(Math.random() * (Math.floor(max) - Math.ceil(min) + 1)) + Math.ceil(min);
    };
    $.ajax({
      url: `/photos/${restId}`,
      method: 'GET',
      success: function(data) {
        for (var i = 0; i < data.length; i++) {
          foodPics.push(data[i].url);
        }
        context.setState({
          images: foodPics,
          name: data[0].name,
          location: data[0].location,
          avatar: data[0].url
        });
      },
      error: function(err) {
        console.log(err);
      }
    });
  }

  nextPhoto() {
    var newArr = this.state.images;
    var hold = newArr.shift();
    newArr.push(hold);
    this.setState({
      images: newArr
    });
  }

  previousPhoto() {
    var newArr = this.state.images;
    var hold = newArr.pop();
    newArr.unshift(hold);
    this.setState({
      images: newArr
    });
  }


  render() {
    return (
      <div>
        <Photos info={this.state} nextPhoto={this.nextPhoto} previousPhoto={this.previousPhoto}/>
      </div>
    );
  }
}

export default App;
