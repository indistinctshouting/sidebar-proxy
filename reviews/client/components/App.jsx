import React from 'react';
import axios from 'axios';
import ReviewList from './ReviewList.jsx';
import styled, { injectGlobal } from 'styled-components';
import reset from 'styled-reset';

injectGlobal`
  ${reset}
`;

const Wrapper = styled.body`
  font-size: 14px;
  line-height: 1.28571em;
  font-family: Helvetica Neue,Helvetica,Arial,sans-serif;
  color: #333;
`;
export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      reviews: []
    };
  }
  
  componentDidMount() {
    let restaurantId = window.location.pathname.split('/')[1];
    this.getReviewData(restaurantId);
  }

  getReviewData(id) {
    axios.get(`/reviews/id/${id}`)
      .then((response) => {
        console.log(response.data);
        this.setState({
          reviews: response.data
        });
      });
  }

  render() {
    return (
      <Wrapper>
        <ReviewList reviews={this.state.reviews}/>
      </Wrapper>
    );
  }
}