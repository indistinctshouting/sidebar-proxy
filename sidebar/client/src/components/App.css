.sidebar {
  position: relative;
  width: 80%;
  margin-left: auto;
  margin-right: auto;
}

.sidebarcontainer {
  display: grid;
  float: right;
  min-width: 330px;
}

.hourcontainer {
  padding-left: 12px;
  justify-self: center;
}