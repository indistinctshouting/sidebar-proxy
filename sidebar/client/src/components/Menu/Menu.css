.modal {
  position: fixed;
  top: 0;
  left: 0;
  width:100%;
  height: 100%;
  background: rgba(36, 35, 35, 0.6);
}

.displayblock {
  display: block;
}

.displaynone {
  display: none;
}

.modalmain {
  position: fixed;
  background: white;
  min-width: 500px;
  min-height: 200px;
  max-height: calc(95vh - 200px);
  width: 50%;
  top:50%;
  left:50%;
  overflow-y: auto;
  transform: translate(-50%,-50%);
  padding: 30px;
  border-radius: 20px
}

.bluetext {
  color: #0073BB;
}

.menuimage {
  border-radius: 10%;
}