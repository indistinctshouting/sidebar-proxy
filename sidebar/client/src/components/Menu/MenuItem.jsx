import React from 'react';
import { Media } from 'react-bootstrap';
import style from './Menu.css';

let emptyImage = "https://s3-media2.fl.yelpcdn.com/assets/2/www/img/dca54f97fb84/default_avatars/menu_medium_square.png";

const MenuItem = (props) => {
  return (
    <div>
      <hr/>
      <Media>
        <Media.Left align="middle">
          <img className={style.menuimage} width={70} height={70} src={emptyImage} alt="thumbnail"/>
        </Media.Left>
        <Media.Body>
          <Media.Heading componentClass="h5" bsClass="pull-left">
            <strong className={style.bluetext}>
              {props.details.item_name}
            </strong>
        </Media.Heading>
          <Media.Heading componentClass="h5" bsClass="pull-right">
            <strong>
              {props.details.item_price}
            </strong>
        </Media.Heading>
          <br></br>
          <br></br>
          <p>{props.details.item_description}</p>
        </Media.Body>
      </Media>
    </div>
  )
}
export default MenuItem
