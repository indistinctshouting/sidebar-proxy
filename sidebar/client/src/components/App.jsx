import React from 'React';
import axios from 'axios';
import Menu from './Menu/Menu.jsx';
import Hours from './Hours/Hours.jsx';
import Summary from './Summary/Summary.jsx';
import styles from './App.css';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showMenu: false,
      restaurantData: {
        menu: {},
        hours: {},
      },
      date: undefined,
    }
    this.showMenu = this.showMenu.bind(this);
    this.hideMenu = this.hideMenu.bind(this);
  }

  showMenu() {
    this.setState({ showMenu: true });
  }

  hideMenu() {
    this.setState({ showMenu: false });
  }

  getData() {
    const id = window.location.pathname.split('/')[2];
    axios.get(`/restaurant/${id}/description`)
      .then((res) => {
        console.log(res.data);
        this.setState({ restaurantData: res.data, date: new Date() });
      });
  }

  componentDidMount() {
    this.getData();
  }


  render() {
    return (
      <div className={styles.sidebar}>
        Yelp
        <br></br>
        <br></br>
        <div className={styles.sidebarcontainer}>
          <div>
            <Summary
              hours={this.state.restaurantData.hours}
              showMenu={this.showMenu}
              priceRange={this.state.restaurantData.price_range}
              health={this.state.restaurantData.health_score}
            />
          </div>
          <div className={styles.hourcontainer}>
            <Hours
              hours={this.state.restaurantData.hours}
            />
          </div>
        </div>
        <Menu
          show={this.state.showMenu}
          hideMenu={this.hideMenu}
          menuInfo={this.state.restaurantData.menu}
          name={this.state.restaurantData.restaurant_name}
        />
      </div>
    )
  }
}

export default App;