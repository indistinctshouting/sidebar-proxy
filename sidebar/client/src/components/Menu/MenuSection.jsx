import React from 'react';
import MenuItem from './MenuItem.jsx'

const MenuSection = (props) => {
  return (
    <div>
      <h4>
        {props.section}
      </h4>
      {props.items.map(item => 
        <MenuItem 
          key={item.item_name} 
          details={item}
        />
      )}
      <br></br>
    </div>
  )
}

export default MenuSection;