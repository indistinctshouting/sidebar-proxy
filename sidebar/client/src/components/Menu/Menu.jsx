import React from 'react';
import MenuSection from './MenuSection.jsx';
import style from './Menu.css';

const Menu = (props) => {
  const showOrHideClass = (props.show) ? `${style.modal} ${style.displayblock}` : `${style.displaynone}`;
  return (
    <div className={showOrHideClass}>
      <section className={style.modalmain}>
        <h4>
          Menu For {props.name}
          <button type="button" className="close" aria-label="Close" onClick={props.hideMenu}>
            <span>&#10005;</span>
          </button>
          <hr />
        </h4>
        {Object.keys(props.menuInfo).map(section =>
          <MenuSection
            key={section}
            section={section}
            items={props.menuInfo[section]}
          />
        )}
        <button onClick={props.hideMenu}>Close</button>
      </section>
    </div>
  )
}

export default Menu;