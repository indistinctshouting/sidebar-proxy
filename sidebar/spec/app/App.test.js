import React from 'react';
import { shallow} from 'enzyme';
import App from '../../client/src/components/App';
import Menu from '../../client/src/components/Menu'

describe('App', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should render App component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a modal menu component', () => {
    expect(wrapper.containsMatchingElement(<Menu />)).toEqual(true);
  });

  it('should have a state property to render modal conditionally', () => {
    expect(typeof wrapper.state('showMenu')).toBe('boolean');
  });

  it('should have a state property to contain restaurant data', () => {
    expect(typeof wrapper.state('restaurantData')).toBe('object');
  });

  it('should set modal to show with the showMenu method', () => {
    const instance = wrapper.instance();
    expect(wrapper.state('showMenu')).toBe(false);
    instance.showMenu();
    expect(wrapper.state('showMenu')).toBe(true);
  });

  it ('should set modal to hide with the hideMenu method', () => {
    const instance = wrapper.instance();
    expect(wrapper.state('showMenu')).toBe(false);
    instance.showMenu();
    instance.hideMenu();
    expect(wrapper.state('showMenu')).toBe(false);
  });

});



