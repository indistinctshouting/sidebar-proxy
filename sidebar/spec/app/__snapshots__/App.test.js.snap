// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App component correctly 1`] = `
<div>
  Yelp
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Show Menu
  </button>
  <br />
  <Menu
    hideMenu={[Function]}
    menuInfo={Object {}}
    show={false}
  />
</div>
`;
